include/float_vector.h

```diff
@@ -42,19 +44,26 @@ namespace quda {
     return c;
   }
 
-  template <typename T> constexpr T zero() { return static_cast<T>(0); }
-  template <> constexpr double2 zero() { return double2 {0.0, 0.0}; }
-  template <> constexpr double3 zero() { return double3 {0.0, 0.0, 0.0}; }
-  template <> constexpr double4 zero() { return double4 {0.0, 0.0, 0.0, 0.0}; }
+  template <typename T> constexpr std::enable_if_t<std::is_arithmetic_v<T>, T> zero() { return static_cast<T>(0); }
```

Comment:
Can you explain what the motivation is for this?  At present this will break run-time compilation of QUDA since `std::is_arithmetic_v` isn't supported in Jitify (I can add it easily enough though if there's a good reason).

Reply:
It's needed to allow non-overlapping versions of the templated function.  This was done to allow the addition of a version for array<T,N>.  There are other ways to do it, but this was the simplest.  We could also just overload for float and double explicitly.

Reply:
That's motivation enough.  I can just added the needed change to Jitify and put this directly into your PR while I'm testing.  Sound ok?